mlx/ops.h

```diff
@@ -110,6 +110,10 @@ inline array identity(int n, StreamOrDevice s = {}) {
   return identity(n, float32, s);
 }
 
+array tri(int n, int m, int k, Dtype type, StreamOrDevice s = {});
```

Comment:
Can you do an overload of  `tri(int n, Dtype ...) where `m=n` and `k=0`. I think it's a common use case.